crates/wac-graph/src/plug.rs

```diff
@@ -0,0 +1,64 @@
+use crate::{
+    types::{Package, SubtypeChecker},
+    CompositionGraph, EncodeOptions, NodeId, PackageId,
+};
+
+/// Take the exports of the plug component and plug them into the socket component.
+pub fn plug(
+    graph: &mut CompositionGraph,
+    plugs: Vec<Package>,
+    socket: PackageId,
+    socket_instantiation: NodeId,
+) -> Result<Vec<u8>, anyhow::Error> {
+    for plug_package in plugs {
+        let plug = graph.register_package(plug_package)?;
+
+        let mut plugs = Vec::new();
```

Comment:
I don't like the shadowing of `plugs` here. Perhaps this can be renamed to `plug_exports`